saleor/graphql/product/utils.py

```diff
@@ -1,45 +1,54 @@
+from collections import defaultdict
+from typing import List
+
 from django.utils.text import slugify
 
-from ...product.models import Attribute
+from ...product import AttributeInputType, models
+
+
+def validate_attribute_input(instance: models.Attribute, values: List[str]):
+    if instance.input_type != AttributeInputType.MULTISELECT and len(values) != 1:
+        raise ValueError(f"A {instance.input_type} attribute must take only one value")
 
 
-def attributes_to_hstore(attribute_value_input, attributes_queryset):
+def attributes_to_json(attribute_value_input, attributes_queryset):
     """Transform attributes to the HStore representation.
 
     Attributes configuration per product is stored in a HStore field as
     a dict of IDs. This function transforms the list of `AttributeValueInput`
     objects to this format.
     """
-    attributes_map = {attr.slug: attr.id for attr in attributes_queryset}
-
-    attributes_hstore = {}
+    attributes_map = {attr.slug: attr for attr in attributes_queryset}
+    attributes_json = defaultdict(list)
 
     values_map = {}
     for attr in attributes_queryset:
         for value in attr.values.all():
             values_map[value.slug] = value.id
 
-    for attribute in attribute_value_input:
-        attr_slug = attribute.get("slug")
+    for attribute_input in attribute_value_input:  # type: dict
```

Comment:
Why do you type `attribute_value_input` here? Don't you think that a better place for that is function parameters?